run_unittests.py

```diff
@@ -759,22 +756,22 @@ def test_find_library_patterns(self):
         else:
             self._test_all_naming(cc, env, patterns, 'linux')
             # Mock OpenBSD since we don't have tests for it
-            true = lambda x, y: True
-            if not is_openbsd():
-                with PatchModule(mesonbuild.compilers.c.for_openbsd,
-                                 'mesonbuild.compilers.c.for_openbsd', true):
-                    self._test_all_naming(cc, env, patterns, 'openbsd')
-            else:
-                self._test_all_naming(cc, env, patterns, 'openbsd')
-            with PatchModule(mesonbuild.compilers.c.for_darwin,
-                             'mesonbuild.compilers.c.for_darwin', true):
-                self._test_all_naming(cc, env, patterns, 'darwin')
-            with PatchModule(mesonbuild.compilers.c.for_cygwin,
-                             'mesonbuild.compilers.c.for_cygwin', true):
-                self._test_all_naming(cc, env, patterns, 'cygwin')
-            with PatchModule(mesonbuild.compilers.c.for_windows,
-                             'mesonbuild.compilers.c.for_windows', true):
-                self._test_all_naming(cc, env, patterns, 'windows-mingw')
+            ###true = lambda env: True
+            ###if not is_openbsd():
+            ###    with PatchModule(mesonbuild.compilers.c.for_openbsd,
+            ###                     'mesonbuild.compilers.c.for_openbsd', true):
+            ###        self._test_all_naming(cc, env, patterns, 'openbsd')
+            ###else:
+            ###    self._test_all_naming(cc, env, patterns, 'openbsd')
+            ###with PatchModule(mesonbuild.compilers.c.for_darwin,
+            ###                 'mesonbuild.compilers.c.for_darwin', true):
+            ###    self._test_all_naming(cc, env, patterns, 'darwin')
+            ###with PatchModule(mesonbuild.compilers.c.for_cygwin,
+            ###                 'mesonbuild.compilers.c.for_cygwin', true):
+            ###    self._test_all_naming(cc, env, patterns, 'cygwin')
+            ###with PatchModule(mesonbuild.compilers.c.for_windows,
+            ###                 'mesonbuild.compilers.c.for_windows', true):
+            ###    self._test_all_naming(cc, env, patterns, 'windows-mingw')
```

Comment:
presumably we need to resolve this?

Reply:
Oh right.... I forgot :(

Reply:
@dcbaker @jpakkane I kind of fell like the right way to do this is to add `[machine]` to the native file, and make these machine file tests. Thoughts?

Reply:
You could also run them as cross tests if that's easier. I think I landed some code for reading cross files in unit tests.